x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -246,5 +270,11 @@ public int hashCode() {
             return result;
         }
     }
+
+    public enum AuthenticationType {
+        REALM,
+        API_KEY,
+        TOKEN
```

Comment:
Is it worth adding `ANONYMOUS` as an option here?
I think the current behaviour will treat it as `REALM` which isn't quite right.
Happy to see it in a follow up PR.

Reply:
Yeah it is. I think there can be some other values as well. Will address in a followup